File: app/src/main/java/com/example/architechturestartercode/datasource/local/MoviesLocalDataSource.java
package com.example.architechturestartercode.datasource.local;

import android.content.Context;

import androidx.lifecycle.LiveData;

import com.example.architechturestartercode.db.AppDatabase;
import com.example.architechturestartercode.model.Movie;

import java.util.List;

public class MoviesLocalDataSource {
    private MoviesDao moviesDao;

    public MoviesLocalDataSource(Context context) {
        this.moviesDao = AppDatabase.getINSTANCE(context).moviesDao();
    }

    public void insertMovie(Movie movie) {
        new Thread(new Runnable() {
            @Override
            public void run() {
                moviesDao.insertMovies(movie);
            }
        }).start();
    }

    public void deleteMovie(Movie movie) {
        new Thread(new Runnable() {
            @Override
            public void run() {
                moviesDao.deleteMovies(movie);
            }
        }).start();

    }

    public LiveData<List<Movie>> getAllMovies() {
        return moviesDao.getAllMovies();
    }
}

File: app/src/main/java/com/example/architechturestartercode/data/movie/datasource/local/MoviesLocalDataSource.kt
package com.example.architechturestartercode.data.movie.datasource.local

import androidx.lifecycle.LiveData
import com.example.architechturestartercode.data.movie.datasource.local.model.LocalMovie
import com.example.architechturestartercode.data.movie.datasource.remote.model.RemoteMovie
import javax.inject.Inject

class MoviesLocalDataSource @Inject constructor(private val moviesDao: MoviesDao) :
    IMoviesLocalDataSource {

    override suspend fun insertMovie(localMovie: LocalMovie) {
        moviesDao.insertMovies(localMovie)
    }

    override suspend fun deleteMovie(localMovie: LocalMovie) {
        moviesDao.deleteMovies(localMovie)
    }

    override fun getAllMovies(): LiveData<List<LocalMovie>> {
        return moviesDao.getAllMovies()
    }
}

